controllers/inferenceservice_controller.go

```diff
@@ -115,31 +115,12 @@ func (r *OpenshiftInferenceServiceReconciler) SetupWithManager(mgr ctrl.Manager)
 		Watches(&source.Kind{Type: &kservev1alpha1.ServingRuntime{}},
 			handler.EnqueueRequestsFromMapFunc(func(o client.Object) []reconcile.Request {
 				r.log.Info("Reconcile event triggered by serving runtime: " + o.GetName())
-				inferenceServicesList := &kservev1beta1.InferenceServiceList{}
-				opts := []client.ListOption{client.InNamespace(o.GetNamespace())}
-
-				// Todo: Get only Inference Services that are deploying on the specific serving runtime
-				err := r.client.List(context.TODO(), inferenceServicesList, opts...)
-				if err != nil {
-					r.log.Info("Error getting list of inference services for namespace")
-					return []reconcile.Request{}
-				}
-
-				if len(inferenceServicesList.Items) == 0 {
-					r.log.Info("No InferenceServices found for Serving Runtime: " + o.GetName())
-					return []reconcile.Request{}
-				}
-
-				reconcileRequests := make([]reconcile.Request, 0, len(inferenceServicesList.Items))
-				for _, inferenceService := range inferenceServicesList.Items {
-					reconcileRequests = append(reconcileRequests, reconcile.Request{
-						NamespacedName: types.NamespacedName{
-							Name:      inferenceService.Name,
-							Namespace: inferenceService.Namespace,
-						},
-					})
-				}
-				return reconcileRequests
+				return r.getReconcileRequestsOnUpdateOfServingRuntime(o)
+			})).
+		Watches(&source.Kind{Type: &networkingv1.NetworkPolicy{}},
+			handler.EnqueueRequestsFromMapFunc(func(o client.Object) []reconcile.Request {
+				r.log.Info("Reconcile event triggered by Network Policy: " + o.GetName())
+				return r.getReconcileRequestsOnUpdateOfServingRuntime(o)
```

Comment:
Is it right to call the `getReconcileRequestsOnUpdateOfServingRuntime` and not the `getReconcileRequestsOnUpdateOfNetworkPolicy` function?

Reply:
yes, you are right. I make the wrong function call. Its not tracked in my testing earlier because both function do similiar task.
I had fixed the function call now.